packages/runtime/runtime-utils/src/summaryUtils.ts

```diff
@@ -59,26 +59,35 @@ export function mergeStats(...stats: ISummaryStats[]): ISummaryStats {
 }
 
 /**
+ * Calculates the byte length of an UTF-8 encoded string
+ * @param str - The string to calculate the byte length of
+ * @returns The byte length of the string
  * @internal
  */
 export function utf8ByteLength(str: string): number {
 	// returns the byte length of an utf8 string
 	let s = str.length;
 	for (let i = str.length - 1; i >= 0; i--) {
-		const code = str.charCodeAt(i);
-		if (code > 0x7f && code <= 0x7ff) {
-			s++;
-		} else if (code > 0x7ff && code <= 0xffff) {
-			s += 2;
-		}
-		if (code >= 0xdc00 && code <= 0xdfff) {
-			i--; // trail surrogate
+		const code = str.codePointAt(i);
+		// Add null check to handle potential undefined
```

Comment:
The comment feels unnecessary (unless there's a special condition that would cause it to be undefined that we need to explain, for example), the code is pretty clear about it.
```suggestion
```